overlord/snapstate/handlers.go

```diff
@@ -752,6 +752,27 @@ func (m *SnapManager) doMountSnap(t *state.Task, _ *tomb.Tomb) error {
 		}
 	}
 
+	// check that there is a "update-gadget-assets" task for kernels too,
+	// see https://bugs.launchpad.net/snapd/+bug/1940553
+	if snapsup.Type == snap.TypeKernel {
+		func() {
+			st.Lock()
+			defer st.Unlock()
+
+			hasGadgetUpdateTask := func() bool {
+				for _, other := range t.Change().Tasks() {
```

Comment:
Fwiw, this is buggy because it does not check that the "update-gadget-assets" task really belongs to the kernel, it could be part of a big refresh that happen to contain the gadget too